src/libraries/Microsoft.Extensions.Logging.Abstractions/ref/Microsoft.Extensions.Logging.Abstractions.cs

```diff
@@ -50,41 +50,41 @@ public partial interface ISupportExternalScope
-        public static void LogError(this Microsoft.Extensions.Logging.ILogger logger, Microsoft.Extensions.Logging.EventId eventId, string message, params object[] args) { }
-        public static void LogError(this Microsoft.Extensions.Logging.ILogger logger, System.Exception exception, string message, params object[] args) { }
-        public static void LogError(this Microsoft.Extensions.Logging.ILogger logger, string message, params object[] args) { }
-        public static void LogInformation(this Microsoft.Extensions.Logging.ILogger logger, Microsoft.Extensions.Logging.EventId eventId, System.Exception exception, string message, params object[] args) { }
-        public static void LogInformation(this Microsoft.Extensions.Logging.ILogger logger, Microsoft.Extensions.Logging.EventId eventId, string message, params object[] args) { }
-        public static void LogInformation(this Microsoft.Extensions.Logging.ILogger logger, System.Exception exception, string message, params object[] args) { }
-        public static void LogInformation(this Microsoft.Extensions.Logging.ILogger logger, string message, params object[] args) { }
-        public static void LogTrace(this Microsoft.Extensions.Logging.ILogger logger, Microsoft.Extensions.Logging.EventId eventId, System.Exception exception, string message, params object[] args) { }
-        public static void LogTrace(this Microsoft.Extensions.Logging.ILogger logger, Microsoft.Extensions.Logging.EventId eventId, string message, params object[] args) { }
-        public static void LogTrace(this Microsoft.Extensions.Logging.ILogger logger, System.Exception exception, string message, params object[] args) { }
-        public static void LogTrace(this Microsoft.Extensions.Logging.ILogger logger, string message, params object[] args) { }
-        public static void LogWarning(this Microsoft.Extensions.Logging.ILogger logger, Microsoft.Extensions.Logging.EventId eventId, System.Exception exception, string message, params object[] args) { }
-        public static void LogWarning(this Microsoft.Extensions.Logging.ILogger logger, Microsoft.Extensions.Logging.EventId eventId, string message, params object[] args) { }
-        public static void LogWarning(this Microsoft.Extensions.Logging.ILogger logger, System.Exception exception, string message, params object[] args) { }
-        public static void LogWarning(this Microsoft.Extensions.Logging.ILogger logger, string message, params object[] args) { }
+        public static System.IDisposable BeginScope(this Microsoft.Extensions.Logging.ILogger logger, string messageFormat, params object?[]? args) { throw null; }
+        public static void Log(this Microsoft.Extensions.Logging.ILogger logger, Microsoft.Extensions.Logging.LogLevel logLevel, Microsoft.Extensions.Logging.EventId eventId, System.Exception? exception, string? message, params object?[]? args) { }
```

Comment:
From looking at the implementation, it does technically look like args can be null, but do we see anyone actually doing that?  I'd be tempted to say that all of these params arrays should be non-nullable (though allowing the objects they contain to be nullable), as that's generally what we've done elsewhere, and explicitly passing in a null array would be discouraged.

Reply:
Agreed. So based on this sample:

https://github.com/dotnet/runtime/blob/4d09ba6b7e458b491d0ba3d8b8d2a0ac0a289a09/src/libraries/Microsoft.Extensions.Logging.Console/tests/Microsoft.Extensions.Logging.Console.Tests/ConsoleLoggerTest.cs#L725-L728

Which would be calling into:

https://github.com/dotnet/runtime/blob/4d09ba6b7e458b491d0ba3d8b8d2a0ac0a289a09/src/libraries/Microsoft.Extensions.Logging.Abstractions/src/LoggerExtensions.cs#L410-L421

Would make sense to make args non-nullable but the the objects themselves be allowed null.

I'll push up a commit shortly to switch all `params object?[]? args` instances in LoggerExtensions.cs to `params object?[] args`